### DISCUSSION TITLE
How to build and run qemu binary on Mac M-series that support hw accelerated graphics with virgl

### DISCUSSION THREAD

**Discussion by mgrossu** [utmapp/UTM#5496] [General]
Is there a tutorial with steps on how to build and run qemu  on Mac M-series that support Linux 3D accelerated guests. The goal is to have  Linux guest with 3D HW accelerated graphics that runs from qemu command line on Mac.

**Comment by macuserguru** [comment]
You mean ASAHI Linux?
ASAHI Linux / docs / SW:Alternative Distros
https://github.com/AsahiLinux/docs/wiki/SW:Alternative-Distros

m-bers / Arch-Linux-Arm-M1
https://github.com/m-bers/Arch-Linux-Arm-M1

Thomas Glanzmann (for Debian)
https://thomas.glanzmann.de/

projects / m1-debian / summary (Thomas Glanzmann)
https://git.zerfleddert.de/cgi-bin/gitweb.cgi/m1-debian/ 

Fedora Asahi Special Interest Group 
https://fedoraproject.org/wiki/SIGs/Asahi

**Reply by mgrossu** [reply]
No, I don't mean Asahi Linux. I mean to build qemu on my Mac, so I can have a qemu binary on my Mac that supports VirGL, so I can run from my terminal `$ qemu-system-aarch64 [...]` and then a Linux VM will start using hardware accelerated graphics with virgl. Thank you.

**Comment by osy** [comment]
https://gist.github.com/akihikodaki/87df4149e7ca87f18dc56807ec5a1bc5

**Reply by mgrossu** [reply]
This is very old. Maybe something newer? Like is it possible to build from UTM repos with qemu 8.0.2 ?  Thank you

**Reply by osy** [reply]
Yeah just build https://github.com/utmapp/qemu but you’ll probably need to follow the linked guide for dependencies and stuff
